src/test/java/io/confluent/connect/elasticsearch/ElasticsearchClientTest.java

```diff
@@ -505,6 +505,41 @@ public void testReporterNotCalled() throws Exception {
     client.close();
   }
 
+  /**
+   * If the record version is set to VersionType.EXTERNAL (normal case for non-streaming),
+   * then same or less
+   * @throws Exception will be thrown if the test fails
+   */
+  @Test
+  public void testExternalVersionConflictReporterNotCalled() throws Exception {
```

Comment:
Should we also test the code path where version type is not EXTERNAL? (data stream or ignore key)

Reply:
+1 if the test isnt already present

Reply:
Note that "data stream" check wasn't until 11.1.x (I did the initial fix on master).  In fact, the data stream config value of "log" seems not introduced until 11.1.x.  But yes, let me look into how to cause this version conflict -- it should be impossible to cause on a blind insert, but maybe not on an read+update, so will probably need to manually read+write+out_of_date_update on that with max retries of 0 (see upsert in DataConverter.java): 
```java
      case UPSERT:
        return new UpdateRequest(index, id)
            .doc(payload, XContentType.JSON)
            .upsert(payload, XContentType.JSON)
            .retryOnConflict(Math.min(config.maxInFlightRequests(), 5));
```

Reply:
@cjolivier01 if its going to make the test or code too complex then we can punt it.

Reply:
did a positive and negative test, the positive test being that it simply hijacks the version type and tests that handleResponse() behaves correctly by reporting the error.